Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -2,11 +2,15 @@
 using System.CommandLine;
 using System.IO;
 using System.Linq;
+using Microsoft.Boogie;
 
 namespace Microsoft.Dafny;
 
 public class CommonOptionBag {
 
+  public static readonly Option<int> ErrorLimit =
+    new("--error-limit", () => 5, "Set the maximum number of errors to report (0 for unlimited).");
```

Comment:
Shouldn't this be called `verification-error-limit`, since it limits only verification errors?

Also, the use-case of the option is unclear to me. If it's only used for this then it would be better if it's a hidden option. What's the reason that the default is not unlimited? Is that to improve performance? If yes, then it might be good to mention performance in the description of this option.

Right now, if I read the description of this option then I'm confused as to why the reported errors are limited to begin with.

Reply:
I also think the default should be unlimited, and that perhaps the option isn't needed at all. My goal for now, though, was to replicate existing functionality in the new CLI, rather than change it. Since the default is to limit the number of errors presented, having an option to raise or eliminate that limit seems important.

Calling it `--verification-error-limit` makes sense. I think I'm going to have another PR to add some more options and could change it then.

Reply:
It can be quite important. If you encode a method as one big VC, then it reports one assertion failure. To get more assertion failures you adjust the VC to prevent the failure you have already seen and rerun. You can continue this until no more failures are found. That iteration takes time and a user might well want to stop after some small number. The problem is that the order in which results are found is non-deterministic. There are arguments for a default either way -- when I was working with production code, I would set it to a small number for a faster debug cycle. No limit only works for users if the results are presented as they are found. I vote to keep it. Like Aaron, I did not know it was only for verification errors.

Reply:
> If you encode a method as one big VC, then it reports one assertion failure. To get more assertion failures you adjust the VC to prevent the failure you have already seen and rerun.

To be clear, I think this is what Boogie does automatically.

> That iteration takes time and a user might well want to stop after some small number.

So it's indeed a performance related option.

> The problem is that the order in which results are found is non-deterministic. 

Why is that a problem?

> I vote to keep it. Like Aaron, I did not know it was only for verification errors.

If we keep it then we should explain why we have it.

Reply:
The non-determinism causes this problem. Suppose you run with an error-limit of 1, find a problem and "fix" it. Then run again. You get a different problem. Did you fix the first one or did you randomly get a different problem?

Non-determinism is also a problem for tests.(And is why I write tests with just one problem, though that does not capture the full range of uses)

And yes, I would count it as a (user) performance optimization.

The user manual says this: limits the number of verification errors reported per procedure. Default is 5; 0 means as many as possible; a small positive number runs faster but a large positive number reports more errors per run

To me that concisely captures the value and tradeoff of the option